Fix: improve height measurement

Gudahtt opened the pull request:
This improves the height measurement when deriving from a string value.
Previously it would work correctly only with pixels, now it also works
with units 'vh' and 'rem'.

Relates to issue #33.

Let me know if there's anything that could be improved - I tried to follow the conventions used elsewhere in the file, but it's likely I missed something.

DataTables commented:
Thanks for this!

A couple of points:

1. I think it would probably be worthwhile having the new function as a new private function (e.g. called `_parseHeight`) as part of the Scroller prototype, rather than an inner function.
1. Since `rem` is being considered, `em` probably should be as well.
1. There is a bit of redundant code - the last two characters being calculated twice for example. I wonder if a simple regex would be easier to read?
1. The radix is used in the first parseInt, but not thereafter. Probably best to be consistent and always use it.
1. A ternary is used in the `parseheight` function, it would probably be worth using one when assigning `heights.viewport` as well since that's a simple condition and assignment.

Hope you don't mind the feedback. It looks like a great addition and I'll be happy to pull it in. Happy to make the changes suggest above, or take an updated PR with them.

Gudahtt commented:
Sure, that all sounds reasonable.

About `em`, I didn't include that one because I wasn't sure how to calculate it correctly, or if it was even possible without... well, doing a lot of work. In order for it to be correct in all cases, we'd have to traverse up the DOM checking font sizing to determine with certainty the font-size of the parent element, then use that to calculate `px` from `em`. Or add something to the DOM and measure it, which... I'm not sure how we'd do hear, because it appears that this function is called before the table gets added to the DOM.

I could just measure the font-size of the `body` element and assume the parent element matches it. That'll probably work fine in *most* cases. I didn't want to take that liberty without your go-ahead though.

Gudahtt commented:
I've updated the PR to address most of these points. Feedback is always welcome - please let me know if I can improve anything further.

I swapped out the `parseInt` for `parseFloat`, as there's no reason those values would have to be integers. It's rounded later in the calculation anyway.

I added support for 'em' as well. Though it's not necessarily correct in all cases, it should work for the vast majority.

DataTables commented:
I think that looks just about perfect. Thanks for making the updates. And I completely agree about the `em` calculation - there is no need to make it absolutely perfect in this case.

The final question is: are you happy for this to be pulled in under the MIT license?

Gudahtt commented:
Absolutely!

DataTables commented:
Awesome - thanks!
